### DISCUSSION TITLE
Queue

### DISCUSSION THREAD

**Discussion by EpicJosch** [BLeeEZ/amperfy#257] [Q&A]
What is the difference between user - and context queue?

**Comment by BLeeEZ** [comment]
Context Queue:
- is the context you pressed play: playlist, album, artists‘ songs
- It will be replaced with the next context when clicking play
- Repeat and shuffle are applied to this queue 

User Queue:
- persists during context changes
- when a song is played it is removed from user queue 

Scenario:
You want to listen to a playlist. Click on play or a song in the playlist. The context queue is filled with the songs of the playlist. 
While listening to the playlist you want to listen to another song. You add it to user queue. It will be played next once. Then you continue to listen to the next context queue song.

**Reply by EpicJosch** [reply]
Wow cool. Thanks for the detailed answer. This is really the ONLY nice and also the best music player app I found on the whole AppStore
